components/resizable_sidebar/resizable_lhs/index.tsx

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {HTMLAttributes, useCallback, useEffect, useMemo, useRef, useState} from 'react';
+import {useSelector} from 'react-redux';
+
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/common';
+
+import {getLhsSize} from 'selectors/lhs';
+import LocalStorageStore from 'stores/local_storage_store';
+import {DEFAULT_LHS_WIDTH, LHS_MIN_MAX_WIDTH} from 'utils/constants';
+import {isOverLimit, isResizableSize, requestAnimationFrameForMouseMove, shouldSnapWhenSizeGrown, shouldSnapWhenSizeShrunk} from '../utils';
+
+interface Props extends HTMLAttributes<'div'> {
+    children: React.ReactNode;
+}
+
+function Resizable({
+    children,
+    id,
+    className,
+}: Props) {
+    const lhsRef = useRef<HTMLDivElement>(null);
+    const resizeLineRef = useRef<HTMLDivElement>(null);
+
+    const [isResizeLineSelected, setIsResizeLineSelected] = useState(false);
+    const previousClientX = useRef(0);
+
+    const lhsSize = useSelector(getLhsSize);
+    const userId = useSelector(getCurrentUserId);
+
+    const minWidth = useMemo(() => LHS_MIN_MAX_WIDTH[lhsSize].min, [lhsSize]);
+    const maxWidth = useMemo(() => LHS_MIN_MAX_WIDTH[lhsSize].max, [lhsSize]);
+
+    const isLhsResizable = useMemo(() => isResizableSize(lhsSize), [lhsSize]);
+
+    const handleDoubleClick = useCallback(() => {
+        if (lhsRef.current) {
+            lhsRef.current.style.width = `${DEFAULT_LHS_WIDTH}px`;
+        }
+
+        LocalStorageStore.setLhsWidth(userId, DEFAULT_LHS_WIDTH);
+    }, [userId]);
+
+    const handleMouseDown = useCallback((e: React.MouseEvent) => {
+        if (!isLhsResizable) {
+            return;
+        }
+
+        previousClientX.current = e.clientX;
+        setIsResizeLineSelected(true);
+        document.body.style.cursor = 'col-resize';
+
+        if (lhsRef.current) {
+            lhsRef.current.classList.add('sidebar--left-dragged');
+        }
+    }, [isLhsResizable]);
+
+    const handleMouseMove = useCallback(requestAnimationFrameForMouseMove((e: MouseEvent) => {
+        if (!lhsRef.current || !previousClientX.current || !resizeLineRef.current) {
+            return;
+        }
+
+        if (!isResizeLineSelected) {
```

Comment:
```ts
   useEffect(() => {
        if (!isResizeLineSelected || !dir) {
            return () => {};
        }

        window.addEventListener('mousemove', handleMouseMove);
```

Before registering the event, we are checking to see if the resize has started. Would this be enough?

Reply:
Yep, I think that should work. That way, we're only registering the `mousemove` handler when the user is dragging the resize line.

We might not actually need to check `dir` there, but I don't think it hurts to include that since that should never actually change